test/e2e/page-objects/benchmark/page-load-benchmark.ts

```diff
@@ -0,0 +1,458 @@
+   * Represents the complete page load time including all resources.
+   */
+  pageLoadTime: number;
+  /**
+   * Time from navigation start to DOM content loaded event end.
+   * Indicates when the initial HTML document has been completely loaded and parsed.
+   */
+  domContentLoaded: number;
+  /**
+   * Time from navigation start to first paint.
+   * The first time any pixel is painted to the screen.
+   */
+  firstPaint: number;
+  /**
+   * Time from navigation start to first contentful paint.
+   * The first time any content (text, images, etc.) is painted to the screen.
+   * This is a key Core Web Vital metric.
+   */
+  firstContentfulPaint: number;
+  /**
+   * Time from navigation start to largest contentful paint.
+   * The time when the largest content element becomes visible.
+   * This is a key Core Web Vital metric for perceived loading speed.
+   */
+  largestContentfulPaint: number;
+  /**
+   * Memory usage statistics (optional, only available in Chrome).
+   * All values are in bytes.
+   */
+  memoryUsage?: {
+    /**
+     * Currently used JavaScript heap size.
+     * Represents the amount of memory actively used by JavaScript objects.
+     */
+    usedJSHeapSize: number;
+    /**
+     * Total allocated JavaScript heap size.
+     * The total amount of memory allocated for the JavaScript heap.
+     */
+    totalJSHeapSize: number;
+    /**
+     * Maximum JavaScript heap size limit.
+     * The maximum amount of memory that can be allocated for the JavaScript heap.
+     */
+    jsHeapSizeLimit: number;
+  };
+};
+
+/**
+ * Individual benchmark measurement result for a single page load test.
+ * Contains the raw performance metrics for one specific test run.
+ */
+export type BenchmarkResult = {
+  /** URL of the page that was tested */
+  page: string;
+  /** Sequential run number for this test iteration */
+  run: number;
+  /** Performance metrics collected during this test run */
+  metrics: BenchmarkMetrics;
+  /** Timestamp when this measurement was taken */
+  timestamp: number;
+};
+
+/**
+ * Statistical summary of benchmark results for a specific page.
+ * Contains aggregated statistics across multiple test runs for performance analysis.
+ */
+export type BenchmarkSummary = {
+  /** URL of the page that was tested */
+  page: string;
+  /** Number of test samples collected for this page */
+  samples: number;
+  /** Mean (average) values for each performance metric */
+  mean: Partial<BenchmarkMetrics>;
+  /** 95th percentile values for each performance metric */
+  p95: Partial<BenchmarkMetrics>;
+  /** 99th percentile values for each performance metric */
+  p99: Partial<BenchmarkMetrics>;
+  /** Minimum values for each performance metric */
+  min: Partial<BenchmarkMetrics>;
+  /** Maximum values for each performance metric */
+  max: Partial<BenchmarkMetrics>;
+  /** Standard deviation values for each performance metric */
+  standardDeviation: Partial<BenchmarkMetrics>;
```

Comment:
mmm we are  getting similar metrics in the `benchmark-mjs` file. Is it the goal to remove the other one eventually?

https://github.com/MetaMask/metamask-extension/blob/main/test/e2e/benchmark.mjs#L18